roles/rsyslog/roles/input_roles/viaq/tasks/main.yaml

```diff
@@ -1,10 +1,9 @@
 ---
-- name: Set Viaq facts
-  set_fact:
-    rsyslog_role_packages: "{{ rsyslog_viaq_prereq_packages|flatten|d([]) }}+{{ rsyslog_viaq_packages|flatten|d([]) }}"
-    rsyslog_role_rules: "{{ rsyslog_viaq_rules }}"
-
+# Deploy configuration files
 - name: Install/Update role packages and generate role configuration files to role subdir
+  vars:
+    __rsyslog_packages: "{{ __rsyslog_viaq_prereq_packages + __rsyslog_viaq_packages }}"
```

Comment:
@nhosoi does this work like this?

Reply:
Yes.  I'm adding a ci test using viaq next...

Reply:
so we can use this instead of `flatten` (re: above comments regarding `flatten`)?

Reply:
Yes, I double-checked by running my test with viaq, which requires 2 package lists and they are combined at the line 5.
```
vars/main.yml:__rsyslog_viaq_prereq_packages: ['nmap-ncat', 'systemd-python', 'policycoreutils', 'checkpolicy', 'policycoreutils-python']
vars/main.yml:__rsyslog_viaq_packages: ['rsyslog-mmnormalize']
```
I experimentally removed rpm packages 'nmap-ncat' and 'rsyslog-mmnormalize' on the target host and ran ansible-playbook to see these 2 packages are installed as expected.

As long as `__rsyslog_.*_prereq_packages` and `__rsylog_.*_packages` are the clean 1 level list, there's no need to have `flatten`.  And lists starting with __ are internal and we manage them, so it's safe to assume they are clean. (if not, it is a bug.)  First I thought having `flatten` is a reasonable safeguard, but I was told to remove them unless they are really necessary, most likely not to hide a hidden bugs.

Reply:
I think even with user supplied values for packages that are supposed to be in `list` format, it is ok to use them with `__default_packages + user_supplied_packages`.  If the user does not provide `user_supplied_packages` in `list` format, they will get an error, which IMO is ok for now. We can go back later and add some sort of input validation if we feel that the default error is not sufficient for some reason.  Also, any such public values should be listed in `defaults/main.yml` and have a default value of `[]`.

Reply:
We already have a user supplied package list `rsyslog_extra_packages`.  It's initialized with [] in roles/rsyslog/defaults/main.yaml.
 roles/rsyslog/defaults/main.yaml
https://github.com/linux-system-roles/logging/pull/87/files#diff-def9bf9161e11520d46eea7aa85093e1R67-R70

And it's flattened before passing to the `package` module.  Hope it's ok...
https://github.com/linux-system-roles/logging/pull/87/files#diff-ed5a08a2e61430dc18e2d3820cbd6f32R3